src/query/interpreter.cpp

```diff
@@ -2874,20 +2875,45 @@ PreparedQuery PrepareEdgeIndexQuery(ParsedQuery parsed_query, bool in_explicit_t
   auto *storage = db_acc->storage();
   auto edge_type = storage->NameToEdgeType(index_query->edge_type_.name);
 
+  std::vector<storage::PropertyId> properties;
+  std::vector<std::string> properties_string;
+  properties.reserve(index_query->properties_.size());
+  properties_string.reserve(index_query->properties_.size());
+  for (const auto &prop : index_query->properties_) {
+    properties.push_back(storage->NameToProperty(prop.name));
+    properties_string.push_back(prop.name);
+  }
```

Comment:
Can we check these properties in cypher main visitor? I would say we should check for number of properties already there

Reply:
This is done the same way as for label-property index. If we are not going to add more properties to indices, I think we can simplify our codebase for several indices.